### PR TITLE
Restructure mappers

### PR DESCRIPTION
**Pull request by lthurston**
This is a draft because I need to test the existing mappers to ensure I didn't impact that inadvertently.

### CODE REVIEW

**Comment by lthurston** [comment]
https://github.com/ucldc/rikolti/issues/219
https://github.com/ucldc/rikolti/issues/218

File: metadata_mapper/mappers/mapper.py [hunk 1]
```diff
@@ -37,62 +68,102 @@ def get_s3_api_response(self):
...
-                if isinstance(value, str):
-                    collated.append(value)
-                else:
-                    collated.extend(value)
+        """multiple field values into a single list
 
-        return collated
+        DEPRECATED: replace with `collate_values()` in mappers that use it
+        """
+        return self.collate_values([self.source_metadata.get(field) for field in fieldlist])
```
Comment (timfrazee) [inline metadata_mapper/mappers/mapper.py:149]: Might be nice to pull in https://pypi.org/project/Python-Deprecated/ and mark these `@deprecated` so we don't forget them.

**Review by timfrazee — APPROVED** [review]

File: metadata_mapper/mappers/mapper.py (hunk 1 above)
Comment (lthurston) [reply metadata_mapper/mappers/mapper.py:149]: Done

**Comment by lthurston** [comment]
@amywieliczka and @barbarahui this PR contains cca_vault and chapman just to simplify things. Although I've done a comparison with Solr for a record for each, I'm sure the validation script will identify more issues with these mappers, and we're prepared to tackle those issues.

**Review by amywieliczka — APPROVED** [review]

**Comment by amywieliczka** [comment]
@lthurston this all looks good to me from an architectural perspective so I've accepted it and personally believe it's ready to merge into main. Once we start getting Solr comparisons really churning, we plan to open new tickets to adjust the specific details of the mappers. Does that work for you?

**Pull request merged**